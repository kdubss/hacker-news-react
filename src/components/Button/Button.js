import React from 'react';

const Button = ({ onClick, className, children }) =>
  <button
    type="button"
    onClick={ () => onClick() }
  >
    { children }
  </button>

export default Button;