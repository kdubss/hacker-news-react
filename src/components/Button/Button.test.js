import React from 'react';
import ReactDOM from 'react-dom';
import Button from './Button';
import renderer from 'react-test-renderer';

describe('Button component tests.', () => {
    it('Button component renders succesfully without crashing', () => {
        const div = document.createElement('div');

        ReactDOM.render(<Button />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    test('Button component has a valid snapshot: ', () => {
        const component = renderer.create(
            <Button>Show Next</Button>
        );
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    })
})


