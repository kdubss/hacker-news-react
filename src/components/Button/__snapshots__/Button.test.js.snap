// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component tests. Button component has a valid snapshot:  1`] = `
<button
  type="button"
>
  Show Next
</button>
`;
